core-processor/src/main/java/io/micronaut/inject/writer/EvaluatedExpressionProcessor.java

```diff
@@ -95,6 +97,11 @@ public void processEvaluatedExpressions(MethodElement methodElement) {
                 return new ExpressionWithContext(expression, evaluationContext);
             })
             .forEach(this::addExpression);
+
+        ClassElement resolvedThis = methodElement.isStatic() || methodElement instanceof ConstructorElement ? null : methodElement.getOwningType();
```

Comment:
can you add a tests:

for  `methodElement.isStatic()` evaluates to true
`methodElement instanceof ConstructorElement` evaluates to true

?